### ISSUE TITLE
General Econometrics task view improvements

### ISSUE THREAD

**Issue by zeileis** [cran-task-views/Econometrics#1]
Welcome Grant @grantmcdermott ! It is great to have you on board for the econometrics task view. It would be great if you could have a look at the task view and check whether there are any gaps or whether you would structure something differently, etc. This is not really urgent and has time until after the christmas break. But maybe you can make some time for it at some point in January. Some helpful comments and pointers are below.

* The general goals and content guidelines are best summarized in the workflow for [proposing new task views](https://github.com/cran-task-views/ctv/blob/main/Proposal.md).
* For the technical guidelines, please install the latest version of the `ctv` package from CRAN and have a look at the [package documentation](https://ctv.r-forge.r-project.org/articles/ctv-howto.html).
* For smaller changes (e.g., adding some packages or links, improving formulations, etc.) feel free to make the changes directly in the [Econometrics.md](../blob/main/Econometrics.md)  task view file.
* For bigger changes (e.g., restructuring the content, adding/dropping entire sections, etc.) it would be best to open an issue here and discuss it first.

**Comment by grantmcdermott** [comment]
Thanks @zeileis. Stoked to be onboard :-) I'll take a look at those links when I get a chance.

**Comment by tappek** [comment]
Thank you for adding! Happy to help! 

Looking at section "Panel Data Models", I suggest:

- Subsection "_Miscellaneous_": add general purpose data manipulation packages `data.table` and `dplyr`, e.g., by changing the the last sentence to "Dedicated fast data preprocessing for panel data econometrics is provided by r pkg("collapse"), multi-purpose data manipulation packages r pkg("data.table") and r pkg("dplyr") can also be used for panel-specific data manipulation." Or even by creating a new sub-section "Data manipulation" and put that sentence in.
- Subsection "_Instrumental variables_" vs. Section "Instrumental variables" - Subsection "_Panel data_": There is overlap between the two (somewhat a necessecity by construction). Reading both, I feel like we miss out on the packages `plm`, `fixest`, `lfe` (and maybe others) which provide IV for various panel IV models (FE, FD, BE, multiple RE methods) and FE, respectively, and seem to be more widely employed than `ivfixed` and `ivpanel`.

**Comment by tappek** [comment]
Given its current content, maybe a better name for sections "Links" could be "Literature", "Special Issues, Books, Manuals", or something similar.

(Currently? 2022-01-17) Broken link for "A Brief Guide to R for Beginners in Econometrics" (https://mondo.su.se/access/content/user/ma@su.se/Public/). Seems like this document is available on M. Arai's ResearchGate profile here: https://www.researchgate.net/publication/239744618_A_Brief_Guide_to_R_for_Beginners_in_Econometrics . However, I feel like this is rather a generel purpse R intro (with some linear regression), so not too much focused on econometrics.

**Comment by tappek** [comment]
Another thought: the file is quite long and one will need to scroll a lot to get an idea what the categories/sections are - could we add a mini-TOC after the intro text?

**Comment by grantmcdermott** [comment]
Seconding everything @tappek  has said. 

Something else along those lines: I don't think the current delineation of `Panel Models` makes complete sense, since we've already mentioned **fixest**, **lfe**, and **alpaca** in the `Microeconometrics` section above. (In turn, these three packages are absent from the panel section, which is where I would naturally look for them.) It's tough because packages don't necessarily fall into distinct categories and some repetition is unavoidable. We struggled with this too for LOST (you can see how we decided to break that up over [here](https://lost-stats.github.io/Model_Estimation/Model_Estimation.html).) I wonder if `Microeconometrics` is too broad of a category?

**Comment by zeileis** [comment]
Great, thanks @tappek and @grantmcdermott for the productive discussion. Some feedback:

* _Arai's web page:_ The web page had more content than just the PDF but the page finally seems to be gone and the links to it in the PDF on ResearchGate are outdated as well. So I just removed that link.
* _Links section:_ The `### Links` section is a special section that is handled by the `ctv` package for consistency. Hence it should not be renamed.
* _Table of contents:_ We have discussed TOCs with the CRAN team. Unfortunately, the CRAN style does not support having something like a floating TOC which I think would be useful for task views. In contrast, I feel including a static (non-floating) TOC at the beginning of the document is more distracting than helpful. Then it takes even longer until you see what's really in the view itself. However, if both of you disagree here and favor adding a TOC, please go ahead and do so.
* _Microeconometrics section:_ It might be a bit broad but originally I wanted to focus on cross-section data models here, mainly those with non-normal responses. If you have a better title for that or would split-up the sections differently, I'm happy to discuss this further.
* _Repetition of packages:_ Repeating packages within the same task view is no problem, I think, especially if the itemized lists are organized by topic (rather than by package as in other task views). So feel free to add such repetitions where you feel they would be appropriate and useful!
* _Panel data & instrumental variables:_ I think it would be easiest if you, @tappek, make a concrete suggestion. If the changes are not too excessive, I'm happy if you just push them to the main branch. Otherwise you can, of course, work on a new branch.
* _General data handling:_ I don't think that this is really an "econometric" topic so I'm not sure whether we really should list `data.table` or `dplyr` as packages in this task view. (If we would, then almost all task views could list them.) I would prefer to link to other resources here rather than including the packages as part of the task view. [_Remark:_ You might argue that the same is true for matrix manipulation packages that we currently also list. But I would argue that many econometrics text books explicitly discuss matrix manipulations and provide equations involving more advanced matrix operations.]

**Comment by tappek** [comment]
Thanks for this!
Some thoughts/suggestions about some of the above points:

- _Table of Contents_: A ToC with 1st level only does not seem too long. [Here ](https://github.com/tappek/Econometrics/blob/main/Econometrics.md) is what it could look like. We could even think of making this shorter by using a table with two columns, filled 50/50, a not so standard ToC. This is a manually maintained ToC as I do not know what standard facilities are supported; given a good stability of the 1st level, manual ToC maintaince would be ok? However, sections' headlines are rendered ugly on GitHub.

- _Panel data & instrumental variables_: Nothing excessive planned, here is proposal: https://github.com/tappek/Econometrics/commit/09a048fad5564e35800d6a1cbc4f9bce6f217998#diff-0cdc2d30d2a4e917ce9bc0f30c786cefa3eda964a8b1458729135d37b2add8c1

- _Links section_: As multiple related links are lumped together in one section and placed at the very bottom of the final page, it is easy to miss these helpful resource (book, manuals etc.). Maybe we could carve out the manuals etc. and put them in a dedicated section? However, it would be a section without R packages.

**Comment by zeileis** [comment]
Thanks, Kevin!

* _Table of contents:_ I don't find it helpful, it interrupts the flow, and a two-column table looks ugly esp. on mobile devices. But if you and @grantmcdermott are very much in favor of it, put it in. Maintenance is not the issue, I think. Rendering on GitHub is also not really relevant. People should look at the HTML version on CRAN anyway.
* _Panel data & instrumental variables:_ Looks good, please go ahead and include it in main.
* _Links section:_ This will not be so far down with the latest layout of the package list (which is not an itemized list anymore), you can try with the R-Forge version. So I think we should keep it where it is.

**Comment by tappek** [comment]
- _Table of contents_: I see your points! The Task View is not that long. No pressure, we can wait and see how the Task View evolves with the new community-based approach and maybe having a ToC will become more relevant in the future, maybe not ;).
- _Panel data & instrumental variables_: done.

**Comment by zeileis** [comment]
Excellent, thanks!

Grant @grantmcdermott , talking about links to other resources: I think you have a very good overview of the useful materials "out there", including your own or those of Andrew Heiss, for example. It would be great if you could adapt the links section accordingly.

**Comment by grantmcdermott** [comment]
For sure. I’ll try to add some tonight.

**Comment by zeileis** [comment]
I had another look at this and I think we have made the immediate changes that we wanted to make and all of us agreed to. Unless one of you @grantmcdermott and @tappek wants to make further modifications at this time, I think we can close the issue.

**Comment by grantmcdermott** [comment]
No, I think we (mostly you) hit the immediate high points. Fine to close on my end. There are few other things I’d like to change / continue to discuss going forward, but will propose those when I have a bit more time once the semester ends.

**Comment by zeileis** [comment]
That's perfectly fine, thanks. So I'll wait for Kevin's feedback and then close probably this issue at this point.

**Comment by tappek** [comment]
Just one more thing about sections IV and/vs. panel (as we started to discuss it here): I realized, I favoured section
_Instrumental variables - Panel data_ over _Panel data models - instrumental variables_ when I extended the first section a bit but now I feel like it should be the other way round (i.e., mention the panel-IV-capable packages in _Panel data models - instrumental variables_ and just refer to that section for panel packages in _Instrumental variables - Panel data_.

Otherwise, perfectly fine for me to close this thread. Made sense to discuss some basic topics combined in this one. For further topics, opening individual issues seems more handy.
